### DISCUSSION TITLE
Prefixed fields in traits

### DISCUSSION THREAD

**Discussion by RazorMeister** [spatie/laravel-data#180] [Ideas]
I've encountered a problem while implementing this package because in request I have field named `additional`. And this field is already used in `AppendableData` trait. (I know that I can Map this field). 

So I suggest to prefix all fields from used traits with for example underscore to avoid such problems. (like additional -> _additional)

**Comment by rubenvanassche** [comment]
You can always send a PR in for this, the problem is that we probably going to need to release a major version since this kinda breaks the API. Some properties can be overwritten on a data object and they will not be used anymore when we prefix them with _.
